presto-spark-base/src/main/java/com/facebook/presto/spark/execution/operator/NativeExecutionOperator.java

```diff
@@ -169,7 +169,9 @@ public Page getOutput()
                     return processResult(page.get());
                 }
                 else {
-                    finished = true;
+                    if (taskInfo.isPresent() && taskInfo.get().getTaskStatus().getState().isDone()) {
```

Comment:
Under what situation will this be false?

Reply:
Task doesn't complete until all processing is completed and all results have been fetched and 'delete-results' call was received.